### PR TITLE
ISPN-8646 PessimisticTxPartitionAndMergeDuringRollbackTest.testDegrad…

### PR DESCRIPTION
**Pull request by pruivo**
…edPartitionWithDiscard failures

https://issues.jboss.org/browse/ISPN-8646

### CODE REVIEW

File: core/src/test/java/org/infinispan/partitionhandling/BaseTxPartitionAndMergeTest.java [hunk 1]
```diff
@@ -358,5 +362,12 @@ public void await(long timeout, TimeUnit timeUnit) throws InterruptedException {
       public void unblock() {
          collection.forEach(BaseTxPartitionAndMergeTest.AwaitAndUnblock::unblock);
       }
+
+      public void disableDiscard() {
+         collection.stream()
+               .map(awaitAndUnblock -> awaitAndUnblock instanceof DiscardFilter ? (DiscardFilter) awaitAndUnblock : null)
```
Comment (rvansa) [inline core/src/test/java/org/infinispan/partitionhandling/BaseTxPartitionAndMergeTest.java]: `.filter(DiscardFilter.class::isInstance).map(DiscardFilter.class::cast)`

File: core/src/test/java/org/infinispan/partitionhandling/BaseTxPartitionAndMergeTest.java [hunk 2]
```diff
@@ -289,16 +291,18 @@ public void unblock() {
 
       private final Class<? extends CacheRpcCommand> aClass;
       private final ReclosableLatch notifier;
+      private volatile boolean discard;
 
       private DiscardFilter(Class<? extends CacheRpcCommand> aClass) {
          this.aClass = aClass;
          notifier = new ReclosableLatch(false);
+         discard = true; //discard everything by default. change if needed in the future.
```
Comment (danberindei) [inline core/src/test/java/org/infinispan/partitionhandling/BaseTxPartitionAndMergeTest.java]: @rvansa I guess the key is that now everything is discarded by default, before a command would be discarded only when `!notifier.isOpened()`

File: core/src/test/java/org/infinispan/partitionhandling/BaseTxPartitionAndMergeTest.java (hunk 1 above)
Comment (pruivo) [reply core/src/test/java/org/infinispan/partitionhandling/BaseTxPartitionAndMergeTest.java]: :+1:

File: core/src/test/java/org/infinispan/partitionhandling/BaseTxPartitionAndMergeTest.java (hunk 2 above)
Comment (pruivo) [reply core/src/test/java/org/infinispan/partitionhandling/BaseTxPartitionAndMergeTest.java]: that was the problem. only the first command is was discarded. The test expects the command only to be executed during the partition merging. So, I discard everything (only a single operation is executed in the tests) and I stop discarding when I merge

File: core/src/test/java/org/infinispan/partitionhandling/BaseTxPartitionAndMergeTest.java (hunk 1 above)
Comment (pruivo) [reply core/src/test/java/org/infinispan/partitionhandling/BaseTxPartitionAndMergeTest.java]: I changed the name to `stopDiscard()`

**Comment by galderz** [comment]
> Test Result (16 failures / +9)

Giving it another CI spin

**Pull request merged**